noise/README.md

```diff
@@ -0,0 +1,400 @@
+to libp2p naming conventions. A valid noise-libp2p protocol ID consists of the
+following elements, separated by `/` characters:
+
+- the string `/noise`
+- the name of the handshake pattern, lowercased
+- the name of the DH function, lowercased
+- the name of the cipher, lowercased
+- the name of the hash function, lowercased
+- the version of the noise-libp2p spec that defines the protocol
+
+If any of the names used to construct the protocol ID contains a `/` character
+(for example, `SHA3/256`), the `/` will be replaced with a `-` to avoid
+ambiguity with the separator character. In practice, none of the currently
+supported functions contain a `/` in their names, so this is likely to remain a
+non-issue.
+
+Using the [supported handshake patterns](#supported-handshake-patterns) and
+[supported DH function, cipher and
+hash](#supported-dh-functions-ciphers-and-hash-functions), the following are
+valid noise-libp2p protocol IDs:
+
+- `/noise/xx/25519/chachapoly/sha256/1.0.0`
+- `/noise/xxfallback/25519/chachapoly/sha256/1.0.0`
+- `/noise/ik/25519/chachapoly/sha256/1.0.0`
+
+The final component of the protocol ID refers to the version of the noise-libp2p
+spec used to implement the protocol and must be `1.0.0`. Future revisions of
+this specification may define new version numbers.
+
+## Static Key Authentication
+
+The [Security Considerations section of the Noise spec][npf-security] says:
+
+    * Authentication: A Noise protocol with static public keys verifies that the
+    corresponding private keys are possessed by the participant(s), but it's up to
+    the application to determine whether the remote party's static public key is
+    acceptable. Methods for doing so include certificates which sign the public key
+    (and which may be passed in handshake payloads), preconfigured lists of public
+    keys, or "pinning" / "key-continuity" approaches where parties remember public
+    keys they encounter and check whether the same party presents the same public
+    key in the future.
+    
+And also:
+
+    * Static key reuse: A static key pair used with Noise should be used with a single
+    hash algorithm. The key pair should not be used outside of Noise, nor with
+    multiple hash algorithms. It is acceptable to use the static key pair with
+    different Noise protocols, provided the same hash algorithm is used in all of
+    them. (Reusing a Noise static key pair outside of Noise would require extremely
+    careful analysis to ensure the uses don't compromise each other, and security
+    proofs are preserved).
+
+All libp2p peers possess a cryptographic keypair which is used to [derive their
+peer id][peer-id-spec], which we will refer to as their "identity keypair." To
+avoid potential static key reuse, and to allow libp2p peers with any type of
+identity keypair to use Noise, noise-libp2p uses a separate static keypair for
+Noise that is distinct from the peer's identity keypair.
```

Comment:
Here is that way I interpret the relationship between the handshake static DH keypair and the libp2p signing keypair.

For each signing identity key, there will be at least be one or more handshake static keys. 

If the application is trying to initiate a connection to a specific identity key, they only determine if they have secure connection to that identity key after a handshake has taken place by receiving a certificate for the static public keys that were used in the connection.

Reply:
This is how I understand it as well. However, "signed identity payload" instead of "certificate" are the words I would lean towards. I believe that the "signed identity payload" may then be used for whichever key type and payload content is sent, which could be one or more certificates or other public key forms.

Reply:
@zmanian yes, I think that sums it up well - I'll try to clarify it some more in the write up.

I like the phrase "signed identity payload", that does seem better than "certificate".